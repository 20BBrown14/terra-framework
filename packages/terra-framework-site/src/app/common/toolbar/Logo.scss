/* stylelint-disable selector-class-pattern */
:local {
  .logo {
    align-items: center;
    color: #fff;
    display: flex;
    flex: 0 0 auto;
  }

  .accessory {
    padding-left: 5px;
  }

  .title {
    overflow: hidden;
    padding-left: 5px;
    padding-right: 4rem;
    text-overflow: ellipsis;
    white-space: nowrap;
    word-break: normal;
    word-wrap: normal;
  }
}
