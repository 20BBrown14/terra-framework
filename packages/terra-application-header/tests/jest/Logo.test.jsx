import React from 'react';

import Logo from '../../src/Logo';

describe('Logo', () => {
  it('should render without optional props', () => {
    const result = shallow((
      <Logo />
    ));
    expect(result).toMatchSnapshot();
  });

  it('should render with provided props', () => {
    const result = shallow((
      <Logo
        accessory={<div>test accessory</div>}
        size="small"
        subtitle="test subtitle"
        title="title"
      />
    ));

    expect(result).toMatchSnapshot();
  });
});
