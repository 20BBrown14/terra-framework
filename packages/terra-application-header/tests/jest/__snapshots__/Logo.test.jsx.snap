// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render with provided props 1`] = `
<div
  className="logo"
>
  <div
    className="accessory"
  >
    <div>
      test accessory
    </div>
  </div>
  <div
    className="title"
  >
    <strong>
      title
    </strong>
    test subtitle
  </div>
</div>
`;

exports[`Logo should render without optional props 1`] = `
<div
  className="logo"
/>
`;
